client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx

```diff
@@ -190,6 +192,35 @@ export default function CampaignItemDetails( props: Props ) {
 		conversion_last_currency_found,
 	} = campaign_stats || {};
 
+	// check if delivery outperformed
+	// for impressions
+	const deliveryEstimates = ( display_delivery_estimate || '' ).split( ':' );
+	let medianImpressions = 0;
+	if ( deliveryEstimates && deliveryEstimates.length >= 2 ) {
+		const [ minImpressions, maxImpressions ] = deliveryEstimates.map( Number );
+		medianImpressions = ( minImpressions + maxImpressions ) / 2;
+	}
+	let impressionsOutperformedPercentage = 0;
+	if ( impressions_total > medianImpressions && medianImpressions > 0 ) {
+		impressionsOutperformedPercentage = Math.round(
+			( ( impressions_total - medianImpressions ) / medianImpressions ) * 100
+		);
+	}
+
+	// for clicks
+	const clickEstimates = ( display_clicks_estimate || '' ).split( ':' );
+	let medianClicks = 0;
+	if ( clickEstimates && clickEstimates.length >= 2 ) {
+		const [ minClicks, maxClicks ] = clickEstimates.map( Number );
+		medianClicks = ( minClicks + maxClicks ) / 2;
+	}
+	let clicksOutperformedPercentage = 0;
+	if ( clicks_total > medianClicks && medianClicks > 0 ) {
+		clicksOutperformedPercentage = Math.round(
+			( ( clicks_total - medianClicks ) / medianClicks ) * 100
+		);
+	}
```

Comment:
I noticed that the used code repeats itself.

It seems that adding something like 
```js
const impressionsOutperformedPercentage = calculateOutperformPercentage(display_delivery_estimate);
const clicksOutperformedPercentage = calculateOutperformPercentage(display_clicks_estimate);
```

and implementing `calculateOutperformPercentage` the way you did is the good one.

Please take with with a grain of salt because the code you've wrote looks great (excepting the repetition itself) and works as expected.

Reply:
you are right.. its better to add a function to calculate the amounts instead of repeating the same code and logic

Reply:
fixed here
https://github.com/Automattic/wp-calypso/pull/97707/commits/0a84ef9baf55a6b3cd23fff4ac41d158d67f35ab